APM-398: Update bat start script to support Jenkins.

tzwickl opened the pull request:
Update bat start script to support Jenkins.

In the Jenkins environment the timeout command was not recognized and thus the script was unusable. The timeout command was replaced with the ping command to provide the same functionality and works now under Windows and in the Jenkins environment.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/inspectit-labs/spring-petclinic-microservices/9)
<!-- Reviewable:end -->

Commit:
APM-398: Update bat start script to support Jenkins.

In the Jenkins environment the timeout command was not recognized and thus the script was unusable. The timeout command was replaced with the ping command to provide the same functionality and works now under Windows and in the Jenkins environment.

ivansenic commented:
When I try to start this in Windows VM I get:

```
"Agent jar found."
"Starting Configuration Server"
C:\Users\Administrator\workspace\spring-petclinic-microservices\spring-petclinic-config-server\configuration-log.txt
The process cannot access the file because it is being used by another process.
The process cannot access the file because it is being used by another process.
"Waiting for Configuration Server"
find: ‘Started ConfigServerApplication’: No such file or directory
find: ‘Started ConfigServerApplication’: No such file or directory
find: ‘Started ConfigServerApplication’: No such file or directory
find: ‘Started ConfigServerApplication’: No such file or directory
find: ‘Started ConfigServerApplication’: No such file or directory
find: ‘Started ConfigServerApplication’: No such file or directory
find: ‘Started ConfigServerApplication’: No such file or directory
find: ‘Started ConfigServerApplication’: No such file or directory
find: ‘Started ConfigServerApplication’: No such file or directory
find: ‘Started ConfigServerApplication’: No such file or directory
find: ‘Started ConfigServerApplication’: No such file or directory
find: ‘Started ConfigServerApplication’: No such file or directory
find: ‘Started ConfigServerApplication’: No such file or directory
find: ‘Started ConfigServerApplication’: No such file or directory
find: ‘Started ConfigServerApplication’: No such file or directory
find: ‘Started ConfigServerApplication’: No such file or directory
find: ‘Started ConfigServerApplication’: No such file or directory
find: ‘Started ConfigServerApplication’: No such file or directory
find: ‘Started ConfigServerApplication’: No such file or directory
find: ‘Started ConfigServerApplication’: No such file or directory
find: ‘Started ConfigServerApplication’: No such file or directory
find: ‘Started ConfigServerApplication’: No such file or directory
find: ‘Started ConfigServerApplication’: No such file or directory
find: ‘Started ConfigServerApplication’: No such file or directory
find: ‘Started ConfigServerApplication’: No such file or directory
find: ‘Started ConfigServerApplication’: No such file or directory
find: ‘Started ConfigServerApplication’: No such file or directory
find: ‘Started ConfigServerApplication’: No such file or directory
find: ‘Started ConfigServerApplication’: No such file or directory
find: ‘Started ConfigServerApplication’: No such file or directory
find: ‘Started ConfigServerApplication’: No such file or directory
find: ‘Started ConfigServerApplication’: No such file or directory
find: ‘Started ConfigServerApplication’: No such file or directory
find: ‘Started ConfigServerApplication’: No such file or directory
find: ‘Started ConfigServerApplication’: No such file or directory
find: ‘Started ConfigServerApplication’: No such file or directory
find: ‘Started ConfigServerApplication’: No such file or directory
find: ‘Started ConfigServerApplication’: No such file or directory
find: ‘Started ConfigServerApplication’: No such file or directory
find: ‘Started ConfigServerApplication’: No such file or directory
find: ‘Started ConfigServerApplication’: No such file or directory
find: ‘Started ConfigServerApplication’: No such file or directory
find: ‘Started ConfigServerApplication’: No such file or directory
find: ‘Started ConfigServerApplication’: No such file or directory
find: ‘Started ConfigServerApplication’: No such file or directory
find: ‘Started ConfigServerApplication’: No such file or directory
find: ‘Started ConfigServerApplication’: No such file or directory
find: ‘Started ConfigServerApplication’: No such file or directory
find: ‘Started ConfigServerApplication’: No such file or directory
find: ‘Started ConfigServerApplication’: No such file or directory
find: ‘Started ConfigServerApplication’: No such file or directory
find: ‘Started ConfigServerApplication’: No such file or directory
find: ‘Started ConfigServerApplication’: No such file or directory
find: ‘Started ConfigServerApplication’: No such file or directory
find: ‘Started ConfigServerApplication’: No such file or directory
find: ‘Started ConfigServerApplication’: No such file or directory
find: ‘Started ConfigServerApplication’: No such file or directory
find: ‘Started ConfigServerApplication’: No such file or directory
find: ‘Started ConfigServerApplication’: No such file or directory
```

any idea what's wrong here?

tzwickl commented:
ok should be fixed with the new commit :)
Problem was that we deleted the log files but already started the mvn process to write to the log file while the deletion process was still ongoing. I now added a short delay between deletion and starting of the mvn command since then I didn't get this message again.
Please check if it works now for you as well.
I also extended the bat script to get a second optional parameter for setting the CMR host.

ivansenic commented:
Seams that it's still the same problem:

```
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 01:23 min
[INFO] Finished at: 2017-11-29T16:19:48+01:00
[INFO] Final Memory: 127M/303M
[INFO] ------------------------------------------------------------------------
"Agent jar found."
"Starting Configuration Server"
Waiting for Configuration Server.find: ‘Started ConfigServerApplication’: No such file or directory
.find: ‘Started ConfigServerApplication’: No such file or directory
.find: ‘Started ConfigServerApplication’: No such file or directory
.find: ‘Started ConfigServerApplication’: No such file or directory
.find: ‘Started ConfigServerApplication’: No such file or directory
.find: ‘Started ConfigServerApplication’: No such file or directory
.find: ‘Started ConfigServerApplication’: No such file or directory
.find: ‘Started ConfigServerApplication’: No such file or directory
.find: ‘Started ConfigServerApplication’: No such file or directory
```
can you re-check, maybe I am doing something wrong :).. Also you added here the optional parameter for the CMR location which should be added as part of #8.. So now it's all mix. Can we maybe move the optional parameter stuff to the #8 and here just keep what's related for the jenkins? Would this work? If not, I could integrate the #8 then after this one is done and working..

tzwickl commented:
Can you show me how you exactly started the batch script and where? What Windows version are you using?
Can you check if the file `spring-petclinic-config-server\configuration-log.txt` is created when starting the batch file? Seems to me as the files aren't created. Maybe due to no write permission?

Regarding the second point, I did this to avoid merge conflicts... Because when I do the changes in #8 we will definitely create some merge conflicts when merging this pull request...

ivansenic commented:
Seams that the problem was that I was starting this in the gitbash.. Seams fine when I start from the Win CMD.. The find commands differ :smile:
